[28기 김정현][Modify] 인증/인가 테스트 완료. Login, Register.js 수정

KimJeongHyun opened the pull request:
## :: 최근 작업 주제 (하나 이상의 주제를 선택해주세요.)
- [x] 기능 추가
- [x] 리뷰 반영
- [x] 리팩토링
- [ ] 버그 수정
- [ ] 컨벤션 수정

<br />

## :: 구현 목표 
1. Cloning instagram's login, main page

2. Need to create other pages

3. Apply floating modal's tooltip

<br />

## :: 구현 사항 설명 (작업한 내용을 상세하게 기록합니다.)

1. 로그인 화면
  - Resizing event : 브라우저 크기에 따라 좌측 imgBox의 표시여부가 결정됨.
  - 테스트 방법 : localhost:3000/login-hyun 접속 후, 브라우저 크기 조정.
  - ID, Password validate에 따른 로그인 버튼 활성화 여부
  - 테스트 방법 : localhost:3000/login-hyun 접속 후, ID 및 PW 입력하여 확인
2. 메인 화면
  - Resizing event : 브라우저 크기에 따라 검색 바와 우측 sideBar의 표시여부가 결정됨.
  - 테스트 방법 : localhost:3000/main-hyun 접속 후, 브라우저 크기 조정.
  - 검색시 결과 필터링
  - 테스트 방법 : localhost:3000/main-hyun 접속 후, 상단 검색바 입력하여 확인.
  - 댓글 달기 기능
  - 테스트 방법 : feed 하단 댓글 input 란에 입력 후 게시 버튼 클릭 또는 엔터키로 확인
  - 댓글 좋아요 기능
  - 테스트 방법 : feed 하단 댓글 box의 하트 모양 버튼 클릭

<br />

## :: 특이 사항

## :: Precautions

1. Semantic Tag의 사용

2. Component 지향적 코드

KimJeongHyun committed:
[Refactor] Main 폴더 리팩토링
 - Semantic Tag 정리
 - addEventListener 최소화 ( window 제외 삭제 )
 - getElementById -> querySelector로 변경
 - FriendFeed scss 변경
 - props destructure 적용

KimJeongHyun committed:
[Modify] Navigation bar css 조정
 - 우측 Nav btn floating box translateX 값조정

KimJeongHyun committed:
[Modify] Login, Main.js 로직 변경
 - eventListener clear 적용
 - input data validation :: Ternary condition operator로 변경

KimJeongHyun committed:
[Modify] MainSideBar useEffect[] 동작 제거
 - 불필요한 Follow 문자열 reduce 함수 제거
 - css text-ellipsis 처리로 ' ... ' 표현 적용됨.

KimJeongHyun committed:
[Add] Notice! Mission 1~4 모두 적용되어 있음.

[Add] 댓글 box 별도 기능 구현
 - 댓글이 2개 이상 있는 경우, View or Hide comment가 활성화됨.
 - 댓글은 최근에 달린 2개까지만 확인되며, view all comments 입력시 모두
   출력됨.
[Modify] remove시 DOM element가 아닌 state를 변경하도록 수정
 - 댓글 컴포넌트에서 각 피드의 commentData 및 setCommentData를 prop으로
   받음
 - 두 prop을 이용해 commentData에서 삭제한 요소를 뺀 리스트로 새로
   업데이트
 - 각 피드 페이지는 map을 돌리고 있으므로 알아서 다시 렌더링된다.

KimJeongHyun committed:
[ADD] Fetch json data 적용
 - 댓글 데이터 json mockup Data에서 받아오도록 수정
 - 댓글 데이터 state 구조 변경
 - 댓글 입/출력 테스트 완료 및 View&hide 정상 확인

KimJeongHyun committed:
[Modify] 댓글 view/hide 로직 변경
 - map 방식에서 filter to map으로 변경됨`

KimJeongHyun committed:
[Modify] event.target 구조분해할당 및 feedData mockup으로 분리 등
 - Login 버튼 클릭 또는 엔터키 입력시 main route
 - event.target 값들 구조분해할당 진행
 - feedData state Mockup data로 불러오도록 수정 및 key:value 구조 수정

KimJeongHyun committed:
[Modify] SearchResultArray.js props 구조 분해할당 진행

KH2Wone commented on src/pages/jeonghyunKim/Login/Login.js:
```diff
@@ -1,3 +1,126 @@
+          loginBtn.style.cursor = 'inherit';
+          loginBtn.setAttribute('disabled', true);
+        })();
+  }, [inputIDValue, inputPWValue]);
```
style 속성은 인라인으로 넣지 않고 className 또는 id를 통해 적용해보는 방식도 좋을 것 같아요 👍🏻
ex) loginBtn.classList.add()

KimJeongHyun replied:
button의 disabled를 state로 관리토록 하여 수정했습니다!

YunKukPark commented on src/pages/jeonghyunKim/Main/Main.js:
```diff
@@ -1,3 +1,75 @@
+  const [feedData, setFeedData] = useState([]);
+
+  useEffect(() => {
+    fetch('http://localhost:3000/data/feedData.json', { method: 'GET' })
+      .then(res => res.json())
+      .then(data => {
+        setFeedData(data);
+      });
```
fetch 의 url 이 반복적으로 일어나면 url을 상수변수로 묶어주면 좋을 것 같습니다 🙌
```jsx
const API_URL = 'http://localhost:3000/data'

fetch(`${API_URL}feedData.json`)
```

KimJeongHyun replied:
유지보수 측면에서 꼭 필요한 코멘트인 듯 합니다👍🏻👍🏻

KimJeongHyun replied:
> 유지보수 측면에서 꼭 필요한 코멘트인 듯 합니다👍🏻👍🏻

별도의 config 파일로 분기해서 가져오는 것으로 변경됐습니다!

![image](https://user-images.githubusercontent.com/23470125/147179910-2c9505b4-b961-4388-9033-d43aa08c684b.png)

seankim1111 reviewed (commented):
고생하셨습니다 정현님! 코드가 깔끔하네요🙂

YunKukPark commented on src/pages/jeonghyunKim/Main/FeedComponent.js:
```diff
@@ -0,0 +1,341 @@
+              View All Comments
+            </div>
+          )}
+          {commentData
+            .filter(function (element) {
+              return viewAllCommentFlag
+                ? element.id >= 0
+                : element.id >= commentData.length - 1;
+            })
+            .map((item, index) => {
+              return (
+                <FeedCommentComponent
+                  id={feedItem.id}
+                  key={index}
+                  commentKey={item.id}
+                  commentEach={item}
+                  commentData={commentData}
+                  setCommentData={setCommentData}
+                />
+              );
+            })}
```
저 이부분 구현이 어려웠는데 참고해가겠습니다 ㅋㅋㅋ 👍🏻

KH2Wone commented on src/pages/jeonghyunKim/Login/Login.js:
```diff
@@ -1,3 +1,126 @@
+  }, [inputIDValue, inputPWValue]);
+
+  const loginFunc = () => {
+    if (inputIDValue.includes('@') && inputPWValue.length >= 5) {
```
`inputIDValue.includes('@') && inputPWValue.length >= 5`
이 부분이 중복되니까 변수로 관리해봐도 좋을 것 같습니다.

KimJeongHyun replied:
boolean return 되는 부가 함수로 check 되도록 통합했습니다!

KH2Wone commented on src/pages/jeonghyunKim/Login/Login.js:
```diff
@@ -1,3 +1,126 @@
+        </section>
+      </article>
+    </main>
+  );
```
시맨틱 태그를 사용하셔서 가독성이 좋아요 🤩

YunKukPark commented on src/pages/jeonghyunKim/Login/Login.js:
```diff
@@ -1,3 +1,126 @@
+  useEffect(() => {
+    const loginBtn = document.querySelector('.loginBtn');
+
+    inputIDValue.includes('@') && inputPWValue.length >= 5
+      ? (() => {
+          loginBtn.style.backgroundColor = 'rgba(var(--d69,0,149,246),1)';
+          loginBtn.style.cursor = 'pointer';
+          loginBtn.removeAttribute('disabled');
+        })()
+      : (() => {
+          loginBtn.style.backgroundColor = 'rgba(var(--d69,0,149,246),.3)';
+          loginBtn.style.cursor = 'inherit';
+          loginBtn.setAttribute('disabled', true);
+        })();
+  }, [inputIDValue, inputPWValue]);
```
JS 를 통해서 스타일을 관리하셨는데 이렇게 해도 좋을 것 같습니다!
```jsx 
const [buttonDisable, setButtonDisable] = useState(true);
// Validate 가 잘 되면 setButtonDisable(false) 로 바꿔줘 
<button className="login-button" type="button" disabled={buttonSwitch} onClick={goToMain} > 로그인 </button>
```
```scss
.login-button {
      margin-top: 10px;
      padding: 8px;
      transition: opacity 0.3s ease-in-out;
      border: 0;
      background-color: $blue;
      color: $white;
      font-size: 14px;
      font-weight: 700;

      &:disabled {
        opacity: 0.3;
        cursor: default;
      }
    }
```


css 에서 disabled 일 때의 opacity와 cursor를 관리하고,
로직은 disabled가 켜졌니 안켜졌니만 신경써주면 되니까 더 깔끔해질 것 같아요!

KimJeongHyun replied:
좋아요! 확인하여 제 방식대로 잘 수정했습니다!

KimJeongHyun replied:
> 좋아요! 확인하여 제 방식대로 잘 수정했습니다!

윤국님께서 하신 것처럼 disabledValue를 따로 state로 빼서 잘 되었습니다

KH2Wone commented on src/pages/jeonghyunKim/Login/login.scss:
```diff
@@ -0,0 +1,159 @@
+  margin: 0;
+  padding: 0;
+  height: 100%;
+}
```
`common.scss`에 나중에 다같이 정리해야해서 건드리지 않으려고 여기 작성하신것같아요!
`reset.scss`, `common.scss` 다같이 병합하고 나서 삭제하시면 되겠네욥 👍🏻

KH2Wone commented on src/pages/jeonghyunKim/Main/FeedCommentComponent.js:
```diff
@@ -0,0 +1,51 @@
+import React from 'react';
+
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
```
리액트에서 컴포넌트로 아이콘을 적용하는 방식이 있었군요, 이거 몰랐어요!!
나중에 다같이 리액트 컴포넌트로 불러올건지, jsx 내에서 i 태그로 적용할건지 의논해봐요 🥳

KH2Wone commented on src/pages/jeonghyunKim/Main/FeedCommentComponent.js:
```diff
@@ -0,0 +1,51 @@
+    setCommentData(updatedCommentList);
+  };
+  return (
+    <span name={commentKey}>
+      <span id="feedCommentsBold">{commentEach.userName}</span>
+      {commentEach.content}
+      {commentEach.isLiked ? (
+        <FontAwesomeIcon
+          className="commentLikeBtn active"
+          icon={faHeart}
+          onClick={likeClick}
+        />
+      ) : (
+        <FontAwesomeIcon
+          className="commentLikeBtn"
+          icon={faHeart}
+          onClick={likeClick}
+        />
+      )}
+      <span id="removeBtn" onClick={removeComment}>
+        삭제
+      </span>
+    </span>
```
`span > span ...` 태그보다는 부모 태그를 `div`로 묶어주거나 그 안에 태그들을 더 가독성 좋은 태그로 적용해보시는걸 추천드려요!!
좋아요 버튼까지 모두 구현 하시다니 많이 하셨네요 멋집니다!!

KH2Wone commented on src/pages/jeonghyunKim/Main/FeedComponent.js:
```diff
@@ -0,0 +1,341 @@
+
+    let inputComments = inputCommentsTextArea.value.replace(/\n/g, '');
+
+    if (inputComments.length > 0) {
```
```suggestion
    if (inputComments.length) {
```
이렇게 해도 좋을듯합니다~!!! 아시다시피 어차피 `length`가 1개 이상이라면 `true`일테니까요.

KimJeongHyun replied:
뭔가 딱딱 떨어지지 않으면 안되는 그런 게 있었어요 ㅠ

수정되었습니다

KH2Wone commented on src/pages/jeonghyunKim/Main/FeedComponent.js:
```diff
@@ -0,0 +1,341 @@
+
+    if (inputComments.length > 0) {
+      setCommentData(current => {
+        const obj = {
```
`obj` 변수명 말고 다른거 어떨까요.... 🧐
객체는 여러개일 수 있지만 저 내용을 담은 객체는 단 하나뿐이니까요.
다른 변수들 네이밍 하신 것처럼 저 객체의 역할을 한번에 알 수 있는 멋진 변수명 지어주세요 ,, ✨

KimJeongHyun replied:
새로운 comment라는 의미로 newComment 로 객체배열 받아오도록 수정했습니다 :)

KH2Wone commented on src/pages/jeonghyunKim/Main/FeedComponent.js:
```diff
@@ -0,0 +1,341 @@
+      <section id="feedComments">
+        <div id="feedCommentsBox" name={'feedCommentsBox' + feedItem.id}>
+          {commentData.length < 3 ? null : viewAllCommentFlag ? (
+            <div className="feedCommentsViewHideBtn" onClick={hideComments}>
+              Hide Comments
+            </div>
+          ) : (
+            <div className="feedCommentsViewHideBtn" onClick={viewAllComments}>
+              View All Comments
+            </div>
```
wow... 댓글 숨김 및 전체보기 기능도 있네요 굿굿

KH2Wone commented on src/pages/jeonghyunKim/Main/MainNavBar.js:
```diff
@@ -0,0 +1,194 @@
+    const searchResultBox = document.querySelector('#searchResultBox');
+    searchIcon.style.display = 'none';
+    searchResultBox.style.display = 'block';
+    event.target.style.paddingLeft = '12px';
```
`scss`파일에서 제어할 수 있도록 하면 좋을 것 같아요.

KimJeongHyun replied:
scss에서 classList add / remove로 제어되도록 변경되었습니다!

KH2Wone commented on src/pages/jeonghyunKim/Main/MainSideBar.js:
```diff
@@ -0,0 +1,89 @@
+            <span id="sideFooterPartText">위치</span>
+            <span id="sideFooterPartText">인기 계정</span>
+            <span id="sideFooterPartText">해시태그</span>
+            <span id="sideFooterPartTextEnd">언어</span>
```
`id`는 중복되면 안돼요. `className`으로 대체하면 더 좋겠습니다!

KimJeongHyun replied:
간혹 생각 없이 이러네요 ㅠ className으로 통합했습니다!

KH2Wone reviewed (commented):
고생 많으셨습니다, 정현님 🔥
많은 것을 구현하셨네요... 멋집니다!!
약간의 리뷰가 있는데 큰 건 아니라서 호로록 읽어보시면 됩니다 👏

KimJeongHyun committed:
[Modify] Login.js input validate function 추가

KimJeongHyun committed:
[Modify] Login.js resizing event modified
 - JS Style control 에서 classList + css 제어를 통해 변경했습니다.

KimJeongHyun committed:
[Modify] Login.js state handler applied
 - state를 한데 모아 handling 합니다.
 - checkInputValidate에서 버튼 disabled 유무를 판단하도록 변경했습니다.

KimJeongHyun committed:
[Modify] Login.js input handling logic modified
 - useEffect로 버튼 disabled 정상적으로 되도록 수정

KimJeongHyun committed:
[Modify] Login.js validate function modify
 - 불필요한 return을 삭제했습니다.

KimJeongHyun committed:
[Add] Register.js 추가
 - 간단한 회원가입 view 추가

KimJeongHyun committed:
[Add] login, register + BE 추가

KimJeongHyun committed:
[Add] wecode test api test clear
 - 로그인, 회원가입, 프로필 조회 api fetch
 - token 저장 및 헤더 포함 완료
 - 결과 all clear

KimJeongHyun committed:
[Refactor] 전체 파일 리팩토링
 - deprecated elements 삭제
 - MainFeedFriends -> MainStories로 변경 및 id, class 재명명
 - inline js control에서 classList 추가/삭제를 통한 css에서의 control
   지향
 - sideBar 중복되는 id -> className으로 변경
 - FeedComponent img sliding 로직 변경
 - 각각 Story list를 mockup에서 가져오도록 변경

KimJeongHyun committed:
[Modify] feed img dot 색상 변화 적용완료
 - onClick에 따라 imgPosition 상태가 변함
 - map 함수의 index가 imgPosition으로 타겟된 경우
 - classList에 active가 있도록 하고, 그 외에는 null 처리로 css에서
   컨트롤

KimJeongHyun committed:
[Refactor] 전체 파일 재 리팩토링
 - Several data가 필요한 경우, DB에서 받아온 것으로 상정해 모두 mockup
   Data화 진행
 - 별도의 config 파일을 통해 fetch url 상수로 제어
 - js, scss 폴더 분리

KimJeongHyun committed:
[Modify] 인증/인가 테스트 완료. Login, Register.js 수정
 - 인증/인가시 알맞은 동작 확인
 - 로그인 / 회원가입시 disabled된 경우에 대한 조건 keyPress function에
   추가됨

KimJeongHyun committed:
.gitignore fix
